src/library_async.js

```diff
@@ -823,6 +881,155 @@ mergeInto(LibraryManager.library, {
+    var fstack_base = stack + stack_size;
+    var fstack_max = stack;
+
+    // init stack pointers
+    {{{ makeSetValue('fib',  4, 'fstack_base', 'i32') }}};
+    {{{ makeSetValue('fib',  8, 'fstack_max',  'i32') }}};
+    {{{ makeSetValue('fib', 12, 'fstack_base', 'i32') }}};
+
+    // init continuation (call user-supplied function on first entry)
+    Fibers.continuations[fib] = function() {
+#if STACK_OVERFLOW_CHECK
+      writeStackCookie();
+#endif
+      {{{ makeDynCall('vi') }}}(funcptr, userdata);
+    };
+
+    return fib;
+  },
+
+  emscripten_fiber_create_from_current_context__sig: 'i',
+  emscripten_fiber_create_from_current_context__deps: ['$Fibers', 'malloc'],
+  emscripten_fiber_create_from_current_context: function() {
+    var fib = _malloc(16);
+
+    {{{ makeSetValue('fib',  0, 0, 'i32') }}};
+    {{{ makeSetValue('fib',  8, 'STACK_MAX',  'i32') }}};
+    {{{ makeSetValue('fib', 12, 'STACK_BASE', 'i32') }}};
+
+    // NOTE: We don't need to initialize the other fields, because swapping will
+    // overwrite them anyway. We can not swap into this fiber without swapping
+    // out of it first anyway, because we'd need a continuation function.
+    //
+    // It may be possible to support this by doing an Asyncify.handleSleep here
+    // just to capture the wakeUp callback, but I'm not sure if it's a useful
+    // thing to support.
+
+    return fib;
+  },
+
+  emscripten_fiber_recycle__sig: 'viii',
+  emscripten_fiber_recycle__deps: ["$Fibers"],
+  emscripten_fiber_recycle: function(fib, funcptr, userdata) {
+    var async_ctx = {{{ makeGetValue('fib', 0, 'i32') }}};
+
+    // this may be undefined when async_ctx == 0, or when recycling the 'current' fiber
+    if (Asyncify.dataInfo[async_ctx] !== undefined) {
+      Asyncify.freeData(async_ctx);
+    }
+
+    {{{ makeSetValue('fib', 0, 0, 'i32') }}};
+
+    var fstack_base = {{{ makeGetValue('fib', 12, 'i32') }}};
+    {{{ makeSetValue('fib', 4, 'fstack_base', 'i32') }}};
+
+    Fibers.continuations[fib] = function() {
+#if STACK_OVERFLOW_CHECK
+      writeStackCookie();
+#endif
+      {{{ makeDynCall('vi') }}}(funcptr, userdata);
+    };
+  },
+
+  emscripten_fiber_free__sig: 'vi',
+  emscripten_fiber_free__deps: ['$Asyncify', '$Fibers', 'free'],
+  emscripten_fiber_free: function(fib) {
+    var async_ctx = {{{ makeGetValue('fib', 0, 'i32') }}};
+
+    // this may be undefined when async_ctx == 0, or when freeing the 'current' fiber
+    if (Asyncify.dataInfo[async_ctx] !== undefined) {
+      Asyncify.freeData(async_ctx);
+    }
+
+    delete Fibers.continuations[fib];
+    _free(fib);
+  },
+
+  emscripten_fiber_swap__sig: 'vii',
+  emscripten_fiber_swap__deps: ["$Asyncify", "$Fibers"],
+  emscripten_fiber_swap: function(f_old, f_new) {
+    return Asyncify.handleSleep(function(wakeUp) {
+      Asyncify.afterUnwind = function() {
+        /*
+         * save caller context
+         */
+
+        var asyncify_context = Asyncify.currData;
```

Comment:
please use `asyncifyContext`, `fOld`, `fNew` (or maybe `newFiler, oldFiber`?) etc. to follow the rest of the JS naming convention.